servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/DefaultMultiAddressUrlHttpClientBuilder.java

```diff
@@ -244,7 +267,184 @@ public StreamingHttpClient apply(final UrlKey urlKey) {
                 singleAddressInitializer.initialize(urlKey.scheme, urlKey.hostAndPort, builder);
             }
 
-            return builder.buildStreaming();
+            StreamingHttpClient singleClient = builder.buildStreaming();
+
+            return new SingleAddressStreamingHttpClientWrapper(singleClient, executionContext.executionStrategy());
+        }
+    }
+
+    private static final class SingleAddressStreamingHttpClientWrapper implements StreamingHttpClient {
+
+        final StreamingHttpClient singleClient;
+        final HttpExecutionStrategy multiClientStrategy;
+
+        SingleAddressStreamingHttpClientWrapper(final StreamingHttpClient singleClient,
+                                                final HttpExecutionStrategy multiClientStrategy) {
+            this.singleClient = singleClient;
+            this.multiClientStrategy = multiClientStrategy;
+        }
+
+        @Override
+        public Completable closeAsync() {
```

Comment:
We have to implement `closeGracefully()` method too

Reply:
The default is defined in `GracefulAutoCloseable` and invokes `close()`. It will result in non-graceful closure if the method is not implemented and delegated.

Reply:
`StreamingHttpClient` also provides a default which does a graceful close. This may be moot if the filter approach is used though.

Reply:
Sorry for confusion, when I posted the original comment, I had `AsyncCloseable#closeAsyncGracefully()` in mind instead of `GracefulAutoCloseable#closeGracefully()`